<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Matthew Stephens" />

<meta name="date" content="2017-02-21" />

<title>Compare bulk vs single</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">singlecell-ideas</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Compare bulk vs single</h1>
<h4 class="author"><em>Matthew Stephens</em></h4>
<h4 class="date"><em>2017-02-21</em></h4>

</div>


<p><strong>Last updated:</strong> 2017-02-21</p>
<p><strong>Code version:</strong> 1a73d0d</p>
<div id="outline" class="section level1">
<h1>Outline</h1>
<p>The purpose here is to outline/brainstorm thoughts on comparing single cell data vs bulk data, and examine the differences between them.</p>
<p>We know from quick plots that the bulk and single cell are “largely concordant”, but can we quantify the differences? Do the differences tend to occur more in the low expressed genes, or are there also differences across the dynamic range?</p>
</div>
<div id="notation-and-model" class="section level1">
<h1>Notation and Model</h1>
<p>Let’s try to set notation and come up with a model.</p>
<p>Start by assuming we have both single cell data and corresponding bulk data on a single sample. Let <span class="math inline">\(X^s_{jg}\)</span> denote the observed counts of gene <span class="math inline">\(g\)</span> in single cell <span class="math inline">\(j\)</span>. And let <span class="math inline">\(X^b_g\)</span> denote the counts of gene <span class="math inline">\(g\)</span> in the bulk.</p>
<p>I’m going to suggest pooling the single cell data into a single count, and define <span class="math inline">\(X^s_g := \sum_j X^s_{jg}\)</span>.</p>
<p>Now we might be interested in identifying the genes that show the most “significant” deviations between <span class="math inline">\(X^b_g\)</span> and <span class="math inline">\(X^s_g\)</span>, and to quantify those deviations.</p>
<p>A simple approach would be to condition on the total counts, <span class="math inline">\(C_g := X^b_g + X^s_g\)</span>, and look for “outliers” where the fraction <span class="math inline">\(X^b_g/C_g\)</span> is particularly small or large. In other words, look for genes where the bulk data reads account for a smaller (or larger) proportion of reads than is typical. The statistical challenge is that we want to take account of the fact that, if <span class="math inline">\(C_g\)</span> is small, then this ratio could be small or big just by chance. See <a href="http://varianceexplained.org/r/empirical_bayes_baseball/">this blog post</a> for relevant discussion.</p>
<p>To account for this we could use a binomial model: <span class="math display">\[X^b_g | C_g \sim \text{Binomial}(C_g, p_g)\]</span> where <span class="math inline">\(p_g\)</span> is the fraction of all reads that come from bulk at gene <span class="math inline">\(g\)</span>, to be estimated. If the bulk data and single-cell data were completely concordant then the <span class="math inline">\(p_g\)</span> would be equal across genes (to some value that is not so important - it would depend on total sequencing depth of bulk vs single-cell). We are interested in variation of <span class="math inline">\(p_g\)</span> among genes - and particularly genes where <span class="math inline">\(p_g\)</span> is particularly large or small.</p>
<p>If we assume that the <span class="math inline">\(p_g\)</span> come from some unimodal distribution <span class="math inline">\(g()\)</span>, which is to be estimated, then we can fit this model using the “general” version of <code>ashr</code>. Alternatively we could potentially use methods from <a href="http://varianceexplained.org/r/empirical_bayes_baseball/">this blog post</a>.</p>
</div>
<div id="sanity-checks" class="section level1">
<h1>Sanity checks</h1>
<p>We could check this approach by applying it also to compare two random subsets of the single cell data. We should find that <span class="math inline">\(p_g \approx 0.5\)</span> for all <span class="math inline">\(g\)</span> in that case.</p>
</div>
<div id="caveatsimprovements" class="section level1">
<h1>Caveats/Improvements</h1>
<p>By pooling the data across single cells this approach obviously ignores the fact that some genes show more variance across single cells than others. We might want to take that into account in the future, depending on the results here.</p>
<p>Actually the question of investigating variability across single cells could itself be potentially interesting. We could fit the model <span class="math inline">\(X^s_{jg} \sim Poi(\mu_{jg})\)</span> where <span class="math inline">\(\log(\mu_{jg}) \sim f_g()\)</span> for some unimodal <span class="math inline">\(f\)</span> (I switched notation from <span class="math inline">\(g\)</span> to <span class="math inline">\(f\)</span> to avoid using <span class="math inline">\(g_g\)</span>.) We could then use <code>ashr</code> to estimate the distribution <span class="math inline">\(f_g()\)</span> for each gene <span class="math inline">\(g\)</span>. I’m interested in whether some <span class="math inline">\(f_g\)</span> have a mode at 0, and others have modes elsewhere. It would also be interesting to see if there is evidence for bimodal <span class="math inline">\(f_g\)</span> (e.g. a mode at 0, and a mode elesehwere), which could be a signature of “bursty” transcription. However, fitting that model might require some methods work.</p>
<div id="session-information" class="section level2">
<h2>Session Information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.2 (2016-10-31)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X El Capitan 10.11.6

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] backports_1.0.5 magrittr_1.5    rprojroot_1.2   htmltools_0.3.5
 [5] tools_3.3.2     whisker_0.3-2   rstudioapi_0.6  yaml_2.1.14    
 [9] Rcpp_0.12.9     stringi_1.1.2   rmarkdown_1.3   knitr_1.15.1   
[13] git2r_0.18.0    stringr_1.1.0   digest_0.6.12   workflowr_0.3.0
[17] evaluate_0.10  </code></pre>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
